### PR TITLE
web: introduce HUDGrid

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu,

Please review the following commits I made in branch nicks/panels:

aaf34c0bfba132400cb8e20c849ac3686f55a1bb (2019-11-04 22:24:18 -0500)
web: introduce HUDGrid
Now, HUDGrid manages the layout and overlap between components, rather than
every component having to know about the other components in the grid.

### CODE REVIEW

**Comment by nicks** [comment]
Today I learned: creating styled-components in the render() method led to major performance problems. They need to live in the global scope. (This isn't so obvious in TFT, which only renders once, but is much more obvious here)

**Comment by nicks** [comment]
(I did some CSS descender tricks to resolve this problem, but it's something we should keep an eye out for on future code reviews)

File: web/src/HUDGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React from "react"
+import styled from "styled-components"
+import { Height, Width, ZIndex } from "./constants"
+
+// The HUD UI looks like this:
+//
+// ----------------------------------
```
Comment (hyu) [inline web/src/HUDGrid.tsx]: YES! ASCII art layout explanations FTW!

File: web/src/HUDGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+`
+
+let MainFrame = styled.div`
+  display: flex;
+  align-items: stretch;
+  width: 100%;
+  padding-bottom: ${Height.statusbar}px;
+  transition: margin-right 300ms ease;
+  padding-top: ${Height.topBar}px;
+  margin-right: ${Width.sidebar}px;
```
Comment (hyu) [inline web/src/HUDGrid.tsx]: Is it problematic to have width 100% AND a right- (or left-) margin? Could it ever create a hard-to-interpret situation for some browsers?

(I honestly am not sure. This is a thing I've done a lot but recently started to suspect)

Comment (nicks) [reply web/src/HUDGrid.tsx]: ya, i think making it padding-right is probably more correct here

**Comment by nicks** [comment]
ya, i agree we need to do a naming pass on the different components.

**Pull request merged**